src/Razor/src/Microsoft.AspNetCore.Razor.LanguageServer/CodeActions/Razor/GenerateMethodCodeActionProvider.cs

```diff
@@ -29,11 +29,11 @@ internal class GenerateMethodCodeActionProvider : IRazorCodeActionProvider
             return s_emptyResult;
         }
 
-        var change = new SourceChange(context.Location.AbsoluteIndex, length: 0, newText: string.Empty);
         var syntaxTree = context.CodeDocument.GetSyntaxTree();
-        var owner = syntaxTree.Root.LocateOwner(change);
+        var owner = syntaxTree.Root.FindToken(context.Location.AbsoluteIndex).Parent;
         if (owner is null)
         {
+            Debug.Fail("Should not be possible for this to be null");
```

Comment:
Is there a reason we can't just do `AssumeNotNull`?

Reply:
Because I'm unfamiliar with the IDE's preference for throwing vs asserting. If you prefer to throw, I'm happy to do that instead.

Reply:
It's unclear to me too. I don't think there's a standard, though we've been moving to `AssumeNotNull()` more recently.

Reply:
Moved to `Assumes.NotNull`.